spec/account/account-manager.spec.js

```diff
@@ -0,0 +1,149 @@
+            "initiateAsync"
+        ]);
+        accountServiceFake.completeAsync = jasmine.createSpy("complete").andCallFake(() => {
+            return promiseMock.resolve(true);
+        });
+        accountServiceFake.getAsync = jasmine.createSpy("get").andCallFake(() => {
+            return promiseMock.resolve({
+                mfaKey: "339r93939303093"
+            });
+        });
+        accountServiceFake.initiateAsync = jasmine.createSpy("initiate").andCallFake((accountId, accountInfo, options) => {
+            return promiseMock.resolve({
+                accountId: accountInfo.accountId,
+                email: accountInfo.email,
+                mfaKey: accountInfo.mfaKey,
+                salt: accountInfo.salt
+            });
+        });
+        hotpFake = jasmine.createSpyObj("hotpFake", [
+            "generateSecretAsync",
+            "verifyToken"
+        ]);
+        hotpFake.generateSecretAsync = jasmine.createSpy("generateSecretAsync").andCallFake(() => {
+            return promiseMock.resolve("thisisasecrectcodefrommfa");
+        });
+        hotpFake.verifyToken = jasmine.createSpy("verifyToken").andCallFake((key, token, opts) => {
+            if (token === "987654") {
+                return false;
+            }
+
+            return true;
+        });
+        config = {
+            account: {
+                completeLifetime: {
+                    months: 6
+                },
+                initiateLifetime: {
+                    hours: 1
+                }
+            },
+            hotp: {
+                previous: {
+                    afterDrift: 1,
+                    beforeDrift: 1
+                }
+            }
+        };
+        accountManager = new AccountManager(accountServiceFake, config, hotpFake, otDateMock, randomMock, promiseMock);
+    });
+    describe(".initiate()", () => {
+        it("gets an object back from service", (done) => {
+            accountManager.initiateAsync({
+                email: "some.one@example.net"
+            }).then((result) => {
+                expect(result).toEqual({
+                    accountId: jasmine.any(Buffer),
+                    email: "some.one@example.net",
+                    mfaKey: "thisisasecrectcodefrommfa",
+                    salt: jasmine.any(Buffer)
+                });
+                expect(result.accountId.length).toBe(24);
+                expect(result.salt.length).toBe(128);
+
+            }).then(done, done);
+        });
+    });
+    describe(".completeAsync()", () => {
+        var expectError, fail;
+
+        // TODO: Create custom matchers
+        function expectError (done, contains) {
+            // Generate a function that asserts the result is an error
+            // and contains some text in the message.
+            return (err) => {
+                expect(err).toEqual(jasmine.any(Error));
+                expect(err.toString()).toContain(contains);
+                done();
```

Comment:
Custom matchers can not call `done` for you.  Maybe a helper function or maybe use a matcher and manually call done.  To be determined LATER, not now.

```
jasmine.expectError(done, contains);
jasmine.fail(done)
```